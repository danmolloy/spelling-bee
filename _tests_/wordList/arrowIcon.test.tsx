import { render, screen, act, fireEvent } from "@testing-library/react"
import "@testing-library/jest-dom"
import ArrowIcon from "../../app/wordList/arrowIcon"

const mockProps = {
  showList: jest.fn()
}

describe("ArrowIcon component", () => {
  beforeEach(() => {
    render(<ArrowIcon {...mockProps} />)
  })
  it("arrow-icon is in the document", () => {
    const arrowIcon = screen.getByTestId("arrow-icon")
    expect(arrowIcon).toBeInTheDocument()
  })
  it("showList() is called on click", () => {
    const arrowIcon = screen.getByTestId("arrow-icon")
    act(() => {
      fireEvent.click(arrowIcon)
    })
    expect(mockProps.showList).toHaveBeenCalled()
  })
  it("matches snapshot", () => {
    const arrowIcon = screen.getByTestId("arrow-icon")
    expect(arrowIcon).toMatchSnapshot()
  })
})