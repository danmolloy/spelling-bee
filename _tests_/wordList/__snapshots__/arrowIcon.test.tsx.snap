// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowIcon component matches snapshot 1`] = `
<button
  aria-label="Show found words"
  class="w-8 h-8 flex justify-center items-center self-start rounded-full hover:bg-gray-100 active:bg-gray-200"
  data-testid="arrow-icon"
  id="show-found-words"
>
  <div
    class="border-black border-r-2 border-b-2 h-3 w-3 self-end rotate-45 mb-3 "
  />
</button>
`;
